internal/controller/utils.go

```diff
@@ -149,6 +149,17 @@ func createOrPatchStatefulSet(ctx context.Context, logger logr.Logger, ec *ecv1a
 		},
 	}
 
+	if len(ec.Spec.EtcdOptions) > 0 {
+		for i := 0; i < len(podSpec.Containers); i++ {
+			if podSpec.Containers[0].Name == "etcd" {
```

Comment:
I wonder if the container name must become a constant. Yes, there are other parts of the code that could benefit from that.

Reply:
It makes sense to have it as a constant, I wasn't sure if doing that would fit into the scope of this PR.

As a side note, I'll be creating an issue soon to look at refactoring how the container is defined for improved testability and cleaner code for things like this feature.